ldclient.go

```diff
@@ -67,6 +67,33 @@ const (
 	migrationVarExFuncName = "LDClient.MigrationVariationCtx"
 )
 
+// dataSystem represents the requirements the client has for storing/retrieving/detecting changes related
+// to the SDK's data model.
+type dataSystem interface {
+	DataSourceStatusBroadcaster() *internal.Broadcaster[interfaces.DataSourceStatus]
+	DataSourceStatusProvider() interfaces.DataSourceStatusProvider
+	DataStoreStatusBroadcaster() *internal.Broadcaster[interfaces.DataStoreStatus]
+	DataStoreStatusProvider() interfaces.DataStoreStatusProvider
+	FlagChangeEventBroadcaster() *internal.Broadcaster[interfaces.FlagChangeEvent]
+
+	// Start starts the data system; the given channel will be closed when the system has reached an initial state
+	// (either permanently failed, e.g. due to bad auth, or succeeded, where Initialized() == true).
+	Start(closeWhenReady chan struct{})
+
+	// Stop halts the data system. Should be called when the client is closed to stop any long-running operations.
+	Stop() error
+
+	// Store returns a read-only accessor for the data model.
+	Store() subsystems.ReadOnlyStore
+
+	// DataAvailability indicates what form of data is available.
+	DataAvailability() datasystem.DataAvailability
+}
```

Comment:
We could have dataSystem embed the `subsystems.ReadOnlyStore` interface instead of returning it. That way, the data system just is-a store. 🤷‍♂️